code/Model/SiteTree.php

```diff
@@ -2711,6 +2712,43 @@ public function getPageLevel() {
 	}
 
 	/**
+	 * @param string $action
+	 * @return Controller
+	 */
+	public function getAssociatedController($action = null) {
```

Comment:
If you do stupid things, expect stupid results :P

---

I'd expect that `setController` sets the record on the controller as part of it.